gubernator/update_config.py

```diff
@@ -17,23 +17,36 @@
 """Updates the Gubernator configuration from the Prow configuration."""
 
 import argparse
+import os
 import yaml
 
-def main(prow_config, gubernator_config):
-    prow_data = yaml.load(open(prow_config))
+def main(prow_config, prow_job_config, gubernator_config):
+    configs = [prow_config]
+    for root, _, files in os.walk(prow_job_config):
+        for name in files:
+            if name.endswith('.yaml'):
+                configs.append(os.path.join(root, name))
+
+    print configs
 
     default_presubmits = set()
-    for job in prow_data['presubmits']['kubernetes/kubernetes']:
-        if job.get('always_run'):
-            default_presubmits.add(job['name'])
+    periodic_names = set()
+    for config in configs:
+        prow_data = yaml.load(open(config))
+
+        if 'presubmits' in prow_data and 'kubernetes/kubernetes' in prow_data['presubmits']:
```

Comment:
I think we could do `prow_data.get('presubmits', [])`  But you know how good my python is ;-)

Reply:
close ... `prow_data.get('presubmits', [])` i think :)

Reply:
I *think* this also works? (assume yaml dump is already all dicts)

Reply:
Yes, let's merge what we've got here :-)